test/constant.cpp

```diff
@@ -49,7 +49,7 @@ int main()
         static_assert(
             std::is_same<
                 decltype(1844674407370955161_c),
-                smath::constant<long long, 1844674407370955161>
+                smath::constant<long, 1844674407370955161>
```

Comment:
This one won't pass on my Windows though. Making a platform-agnostic check without warnings might be hard.